bin/i18n/sync-codeorg-in.rb

```diff
@@ -32,6 +33,26 @@ def sanitize(string)
   return string.gsub(/\r(\n)?/, "\n")
 end
 
+# Pull in various fields for custom blocks from .json files and save them to
+# blocks.en.yml.
+def localize_block_content
+  blocks = {}
+
+  Dir.glob('dashboard/config/blocks/**/*.json').sort.each do |file|
+    name = File.basename(file, '.*')
+    config = JSON.parse(File.read(file))['config']
+    blocks[name] = {
+      'text' => config['blockText'],
+    }
+
+    next unless config['args']
+    args_with_options = config['args'].map {|arg| arg['options']}.compact.flatten(1)
+    blocks[name]['options'] = args_with_options.map(&:reverse).to_h unless args_with_options.empty?
```

Comment:
is this going to be sufficient information to consume these strings at the other end? Seems like we would want this to be something like `blocks[name]['options'][arg['name']]`

Reply:
I'm mixing all the options together into a single hash.  So if two dropdowns in a block both have "north" as a key, they'll have to translate the same way in both dropdowns.  But a different block could have "north" in a dropdown translate to something else.

Does that seem like the right level of granularity?  I could make the keys more specific, so each option is grouped under a key for the name of the arg (what you said).  Or I could make the keys even less specific, so "north" as a dropdown option anywhere always translates to the same string.

What do you think is best?  Thanks!

Reply:
I'm more worried about the array values, which are keyed simply by index

Reply:
oh! I just assumed that was an array; my bad.

Even so, it does seem like there is potential for collision, albiet a small potential.

Reply:
Another option would be to prefix each option with the argument name.  Though I think that'll make it harder to parse back in at the other end.

```yaml
Dancelab_addBehavior:
  text: sprite {SPRITE} begins {BEHAVIOR}
Dancelab_atTimestamp:
  text: at {TIMESTAMP} seconds
Dancelab_changeColorBy:
  text: change {COLOR} {METHOD} by {AMOUNT}
  options:
    METHOD "hue": hue
    METHOD "saturation": saturation
    METHOD "brightness": brightness
Dancelab_changeMove:
  text: change {SPRITE} dance move to {MOVE}
  options:
    MOVE 0: Standing
    MOVE 1: Clap
    MOVE 2: Spin
    MOVE 3: Uprock
    MOVE 4: Bboy
    MOVE "next": Next
    MOVE "rand": Random
Dancelab_changePropBy:
  text: change {SPRITE} {PROPERTY} by {VAL}
  options:
    PROPERTY "scale": size
    PROPERTY "rotation": rotation
    PROPERTY "x": x position
    PROPERTY "y": y position
    PROPERTY "direction": movement direction
    PROPERTY "tint": tint
```

Reply:
What does the "redacted" version of this look like?  I was thinking `change {COLOR} {METHOD} by {AMOUNT}` would redact to `change {0} {1} by {2}` to make it clear that COLOR, METHOD, AMOUNT shouldn't be translated.  Does that match what you're thinking?